oshmem/include/pshmem.h

```diff
@@ -1578,18 +2992,189 @@ OSHMEM_DECLSPEC  int pshmem_ptrdiff_test(volatile ptrdiff_t *addr, int cmp, ptrd
+OSHMEM_DECLSPEC  size_t pshmem_longlong_test_some_vector(volatile long long *ivars, size_t nelems, size_t *indices, const int *status, int cmp, long long *values);
+OSHMEM_DECLSPEC  size_t pshmem_uint_test_some_vector(volatile unsigned int *ivars, size_t nelems, size_t *indices, const int *status, int cmp, unsigned int *values);
+OSHMEM_DECLSPEC  size_t pshmem_ulong_test_some_vector(volatile unsigned long *ivars, size_t nelems, size_t *indices, const int *status, int cmp, unsigned long *values);
+OSHMEM_DECLSPEC  size_t pshmem_ulonglong_test_some_vector(volatile unsigned long long *ivars, size_t nelems, size_t *indices, const int *status, int cmp, unsigned long long *values);
+OSHMEM_DECLSPEC  size_t pshmem_int32_test_some_vector(volatile int32_t *ivars, size_t nelems, size_t *indices, const int *status, int cmp, int32_t *values);
+OSHMEM_DECLSPEC  size_t pshmem_int64_test_some_vector(volatile int64_t *ivars, size_t nelems, size_t *indices, const int *status, int cmp, int64_t *values);
+OSHMEM_DECLSPEC  size_t pshmem_uint32_test_some_vector(volatile uint32_t *ivars, size_t nelems, size_t *indices, const int *status, int cmp, uint32_t *values);
+OSHMEM_DECLSPEC  size_t pshmem_uint64_test_some_vector(volatile uint64_t *ivars, size_t nelems, size_t *indices, const int *status, int cmp, uint64_t *values);
+OSHMEM_DECLSPEC  size_t pshmem_size_test_some_vector(volatile size_t *ivars, size_t nelems, size_t *indices, const int *status, int cmp, size_t *values);
+OSHMEM_DECLSPEC  size_t pshmem_ptrdiff_test_some_vector(volatile ptrdiff_t *ivars, size_t nelems, size_t *indices, const int *status, int cmp, ptrdiff_t *values);
+#if OSHMEM_HAVE_C11
+#define pshmem_test_some_vector(ivars, nelems, indices, status, cmp, values)\
+    _Generic(&*(ivars),                                                     \
+        short*:                pshmem_short_test_some_vector,               \
+        unsigned short*:       pshmem_ushort_test_some_vector,              \
+        int*:                  pshmem_int_test_some_vector,                 \
+        long*:                 pshmem_long_test_some_vector,                \
+        long long*:            pshmem_longlong_test_some_vector,            \
+        unsigned int*:         pshmem_uint_test_some_vector,                \
+        unsigned long*:        pshmem_ulong_test_some_vector,               \
+        unsigned long long*:   pshmem_ulonglong_test_some_vector)(ivars, nelems, indices, status, cmp, values)
+#endif
+
 /*
  * Barrier sync routines
  */
 OSHMEM_DECLSPEC  void pshmem_barrier(int PE_start, int logPE_stride, int PE_size, long *pSync);
 OSHMEM_DECLSPEC  void pshmem_barrier_all(void);
-OSHMEM_DECLSPEC  void pshmem_sync(int PE_start, int logPE_stride, int PE_size, long *pSync);
+OSHMEM_DECLSPEC  void pshmem_sync_deprecated(int PE_start, int logPE_stride, int PE_size, long *pSync);
```

Comment:
does it break backward compatibility? programs compiled with older oshmem may fail to run due to missing dependency to ```pshmem_sync```. is it ok?

Reply:
yes it does, but that's expected since we're trying to align to the spec on a major version release

Reply:
Considering c11, for this specfic case, shouldn't we be backward compatible? Please consider the following macro:
`````````
#if OSHMEM_HAVE_C11
#define pshmem_sync(...)                                               \
    _Generic((__OSHMEM_VAR_ARG1(__VA_ARGS__)),                         \
            shmem_team_t: pshmem_team_sync,                            \
            int:          pshmem_sync_deprecated)(__VA_ARGS__)
#endif
`````````